src/utils.ts

```diff
@@ -435,3 +436,18 @@ export async function* combineAsyncIterables<T>(
 	}
 	return;
 }
+
+export function findAFileWithExt(dir: string, exts: string[]): string | null {
+	const entries = readdirSync(dir, { withFileTypes: true });
+	for (const entry of entries) {
+		const fullPath = path.join(dir, entry.name);
+		if (entry.isFile() && exts.includes(path.extname(fullPath))) {
+			return fullPath;
+		}
+		if (entry.isDirectory()) {
+			const file = findAFileWithExt(fullPath, exts);
+			if (file) return file;
+		}
+	}
+	return null;
+}
```

Comment:
this will become trivial when we drop support for node 20 with `glob` https://nodejs.org/docs/latest-v22.x/api/fs.html#fspromisesglobpattern-options